File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/model/JobId.java
package ru.fix.distributed.job.manager.model;

import java.util.Objects;

public class JobId {

    private String id;

    public JobId(String id) {
        Objects.requireNonNull(id);
        this.id = id;
    }

    public String getId() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        JobId jobId = (JobId) o;
        return Objects.equals(id, jobId.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return "Job[" + id + ']';
    }
}

File: distributed-job-manager/src/main/kotlin/ru/fix/distributed/job/manager/model/JobId.kt
package ru.fix.distributed.job.manager.model

import java.util.*

class JobId(
        val id: String
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as JobId

        return id == other.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }

    override fun toString(): String {
        return "Job[$id]"
    }
}
